tycho-indexer/src/extractor/dynamic_contract_indexer/euler/metadata_generator.rs

```diff
@@ -0,0 +1,258 @@
+use tycho_common::models::{blockchain::Block, protocol::ProtocolComponent, TxHash};
+
+use crate::extractor::dynamic_contract_indexer::component_metadata::{
+    MetadataError, MetadataRequest, MetadataRequestGenerator, MetadataRequestType, RpcTransport,
+};
+
+pub struct EulerMetadataGenerator {
+    rpc_url: String,
+}
+
+impl EulerMetadataGenerator {
+    pub fn new(rpc_url: String) -> Self {
+        Self { rpc_url }
+    }
+}
+
+impl MetadataRequestGenerator for EulerMetadataGenerator {
+    fn generate_requests(
+        &self,
+        component: &ProtocolComponent,
+        block: &Block,
+    ) -> Result<Vec<MetadataRequest>, MetadataError> {
+        let mut requests = vec![];
+
+        let balance_transport = RpcTransport::new(
+            self.rpc_url.clone(),
```

Comment:
I'm wondering if it makes sense to have the RPC URL tied to the RpcTransport or if it should be controlled by the RPC Provider.
I'm fine to have it like this for now but I'd like to hear your opinion on the design here.